src/targets/gpu/prefuse_ops.cpp

```diff
@@ -296,38 +266,53 @@ struct gpu_gqa_rotary_embedding : base_group_query_attention
 };
 MIGRAPHX_REGISTER_OP(gpu_gqa_rotary_embedding);
 
-struct gpu_gqa_softmax : base_group_query_attention
+struct gpu_concat_past_present_k : base_group_query_attention
 {
-    std::string name() const { return "gpu::gqa_softmax"; }
+    std::string name() const { return "gpu::concat_past_present_k"; }
 
-    shape compute_shape(std::vector<shape> inputs) const { return inputs.at(2); }
+    shape compute_shape(std::vector<shape> inputs) const { return inputs.back(); }
 };
-MIGRAPHX_REGISTER_OP(gpu_gqa_softmax);
+MIGRAPHX_REGISTER_OP(gpu_concat_past_present_k);
 
-struct gpu_concat_past_present : base_group_query_attention
+struct gpu_concat_past_present_v : base_group_query_attention
 {
-    std::string name() const { return "gpu::concat_past_present"; }
+    std::string name() const { return "gpu::concat_past_present_v"; }
 
-    shape compute_shape(std::vector<shape> inputs) const { return inputs[0]; }
+    shape compute_shape(std::vector<shape> inputs) const { return inputs.back(); }
 };
-MIGRAPHX_REGISTER_OP(gpu_concat_past_present);
+MIGRAPHX_REGISTER_OP(gpu_concat_past_present_v);
```

Comment:
Can you just have one `gpu_concat_past_present`? Both of these ops are exactly the same.